mobify/platform-scaffold
# Implement pagination

Added pagination to Merlins and SFCC.

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1353
 **Linked PRs**: https://github.com/mobify/progressive-web-sdk/pull/721

## Changes
- Added pagination component to `ProductListContent`
- Added URL query keys: `p`, `sort` and filtering option keys
Example URL: `https://www.merlinspotions.com/potions.html?p=2&sort=price`
- `sortOptions` key is added to redux store: `category` branch
- `filterOptions` key is added to redux store: `category` branch
- Four sorting options added to SFCC, see screenshot below
- Enable filtering for SFCC
- Move some functions from scaffold to SDK


## Todos
- [x] Fix filtering for pagination
- [x] Test pagination on search result page on Merlins:
~Not enough products that has more than 1 page in search result...:(~

## Known Bug
- [ ] On Merlins connector, cannot sort list by URL, i.e. `https://www.merlinspotions.com/potions.html?p=2&sort=price` won't sort the list, you have to manually choose sort option in the dropdown list.

## How to test-drive this PR
- Link SDK branch `pagination`
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Go to potions product list page, verify the pagination is working, verify the URL is syncing correctly with the UI, verify sort is working correctly
- Do the same steps for SFCC
- On the left of app header, click `search`, search any products you like, verify pagination works in search results page

## Screenshots
<img width="1271" alt="screen shot 2017-06-27 at 3 44 07 pm" src="https://user-images.githubusercontent.com/10948652/27613473-8a7675ea-5b50-11e7-8ae6-a4512b187e6b.png">

![addtext_com_mtg1ntqzmty2mq](https://user-images.githubusercontent.com/10948652/27613572-1e29ee34-5b51-11e7-832f-321fe6b95d9c.png)
